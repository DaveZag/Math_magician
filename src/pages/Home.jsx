import calculating from '../assets/calculating.png';

const Home = () => (
  <section className="home-section">
    <div className="home-cont flex flex-col">
      <header>
        <h2 className="welcome-txt">Welcome To our page!</h2>
      </header>
      <div className="content flex">
        <p className="home-text">
          This webpage is an application that allows the user to do most
          arithmetic operations. Also, it provides a section with random
          interesting quotes from people who have marked their generations with
          their intelligence, inventions and other deeds.
        </p>
        <img
          className="home-img"
          src={calculating}
          alt="man seated doing math"
        />
      </div>
    </div>
  </section>
);

export default Home;
