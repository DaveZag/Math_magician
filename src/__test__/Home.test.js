import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '../pages/Home';

describe('Tests for the Home page', () => {
  it('Should render the home page properly', () => {
    const homePage = render(<Home />);
    expect(homePage).toMatchSnapshot();
  });

  it('should contain the calculating.png image in the hero section', () => {
    render(<Home />);
    const heroImage = screen.getByAltText('man seated doing math');
    expect(heroImage).toBeInTheDocument();
  });
});
