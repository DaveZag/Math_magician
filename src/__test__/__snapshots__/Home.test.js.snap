// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Home page Should render the home page properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="home-section"
      >
        <div
          class="home-cont flex flex-col"
        >
          <header>
            <h2
              class="welcome-txt"
            >
              Welcome To our page!
            </h2>
          </header>
          <div
            class="content flex"
          >
            <p
              class="home-text"
            >
              This webpage is an application that allows the user to do most arithmetic operations. Also, it provides a section with random interesting quotes from people who have marked their generations with their intelligence, inventions and other deeds.
            </p>
            <img
              alt="man seated doing math"
              class="home-img"
              src="calculating.png"
            />
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="home-section"
    >
      <div
        class="home-cont flex flex-col"
      >
        <header>
          <h2
            class="welcome-txt"
          >
            Welcome To our page!
          </h2>
        </header>
        <div
          class="content flex"
        >
          <p
            class="home-text"
          >
            This webpage is an application that allows the user to do most arithmetic operations. Also, it provides a section with random interesting quotes from people who have marked their generations with their intelligence, inventions and other deeds.
          </p>
          <img
            alt="man seated doing math"
            class="home-img"
            src="calculating.png"
          />
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
